liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PluginConfigSupport.java

```diff
@@ -394,6 +395,14 @@ protected ServerConfigDocument getServerConfigDocument(CommonLoggerI log, File s
         return scd;
     }
 
+    protected ServerConfigDocument getServerConfigDocument(CommonLoggerI log, File serverXML, Map<String, File> libertyDirPropertyFiles) throws IOException {
+        if (scd == null || !scd.getServerXML().getCanonicalPath().equals(serverXML.getCanonicalPath())) {
+            scd = new ServerConfigDocument(log, libertyDirPropertyFiles);
```

Comment:
This is interesting. Is this because of how our tests are setup/run that this is necessary? Also, I see all builds failing. Do you know the cause? I'm not going to dive into the logs at this point and try to figure it out myself.

Reply:
The tests are passing in latest run, but I would still like to understand this change and why it was necessary.

Reply:
This logic was introduced by this [PR](https://github.com/OpenLiberty/ci.maven/pull/1786), which was on the subject of thread safety. I think the move from static makes this piece not necessary.